x-pack/legacy/plugins/ml/public/application/components/navigation_menu/top_nav/top_nav.tsx

```diff
@@ -40,9 +41,17 @@ function updateLastRefresh(timeRange: OnRefreshProps) {
 
 export const TopNav: FC = () => {
   const { chrome, timefilter, timeHistory } = useUiContext();
+  const [globalState, setGlobalState] = useUrlState('_g');
   const getRecentlyUsedRanges = getRecentlyUsedRangesFactory(timeHistory);
 
-  const [refreshInterval, setRefreshInterval] = useState(timefilter.getRefreshInterval());
+  const [refreshInterval, setRefreshInterval] = useState(
+    globalState?.refreshInterval ?? timefilter.getRefreshInterval()
+  );
+  useEffect(() => {
+    setGlobalState({ refreshInterval });
+    timefilter.setRefreshInterval(refreshInterval);
+  }, [refreshInterval?.pause, refreshInterval?.value]);
```

Comment:
as I see `refreshInterval` is immutable, so you probably can just pass it as is?

Reply:
`refreshInterval` can get updated via appState or date picker. Comparing against the full object resulted in multiple updates with the same values in my tests. (Funny enough I even got multiple updates with a `JSON.stringify` because the object could get serialized with differently ordered keys depending on where the update was triggered).